src/system/applications/combat/combat-tracker.ts

```diff
@@ -0,0 +1,151 @@
+import CosmereCombatant from './combatant';
+
+export default class CosmereCombatTracker extends CombatTracker {
+    //overrides default tracker template to implement slow/fast buckets and activation button.
+    //eslint-disable-next-line -- eslint rules want readonly field
+    get template() {
+        return 'systems/cosmere-rpg/templates/combat/combat-tracker.hbs';
+    }
+
+    //modifies data being sent to the combat tracker template to add turn speed, type and activation status and splitting turns between the initiative phases.
+    async getData(
+        options?: Partial<ApplicationOptions> | undefined,
+    ): Promise<object> {
+        const data = (await super.getData(options)) as {
+            turns: CosmereTurn[];
+            [x: string]: unknown;
+        };
+        data.turns = data.turns.map((turn) => {
+            const combatant: CosmereCombatant =
+                this.viewed!.getEmbeddedDocument(
+                    'Combatant',
+                    turn.id,
+                    {},
+                ) as CosmereCombatant;
+            const newTurn: CosmereTurn = {
+                ...turn,
+                turnSpeed: combatant.getFlag(
+                    'cosmere-rpg',
+                    'turnSpeed',
+                ) as string,
+                type: combatant.actor.type,
+                activated: combatant.getFlag(
+                    'cosmere-rpg',
+                    'activated',
+                ) as boolean,
+            };
+            //strips active player formatting
+            newTurn.css = '';
+            return newTurn;
+        });
+        data.fastPlayers = data.turns.filter((turn) => {
+            return turn.type == 'character' && turn.turnSpeed == 'fast';
+        });
+        data.slowPlayers = data.turns.filter((turn) => {
+            return turn.type == 'character' && turn.turnSpeed == 'slow';
+        });
+        data.fastNPC = data.turns.filter((turn) => {
+            return turn.type == 'adversary' && turn.turnSpeed == 'fast';
+        });
+        data.slowNPC = data.turns.filter((turn) => {
+            return turn.type == 'adversary' && turn.turnSpeed == 'slow';
+        });
+
+        return data;
+    }
+
+    override activateListeners(html: JQuery<HTMLElement>): void {
+        super.activateListeners(html);
+        html.find('.cosmere-turn-speed-control').on(
+            'click',
+            this._onClickToggleTurnSpeed.bind(this),
+        );
+        html.find('.cosmere-activate-control').on(
+            'click',
+            this._onActivateCombatant.bind(this),
+        );
+    }
+
+    //toggles combatant turn speed on clicking the "fast/slow" text on the combat tracker window
```

Comment:
This can also be set up as a doc comment

Reply:
Not exactly sure what you mean here. Can you elaborate a bit?

Reply:
If you use JSDoc comments (comments starting with `/**`) instead of regular comments, then your comment will show up alongside intellisense on most IDEs. Have a look at the Actor document for some examples.

Broadly: Any comments that explain the functionality (for someone who might want to use that function/variable/parameter/etc.) should be doc comments. Any comments that explain the implementation should be regular comments.

Reply:
This was a pretty good explanation. I hadn't used JSDoc comments before so I didn't realize they did that for intellisense. I've expanded the comments in the code I've added.